File: front/scada/src/app/models/Alarm.ts
export interface Alarm{
    id?:string,
    type:string,
    threshold?:number,
    message?:string,
    timeOfActivation?:Date,
    timeStamp?:Date,
    measureUnit?:string,
    priority?:number,
    analogId?:number
}
export interface ActivatedAlarm{
  id?:number,
  timestamp:Date,
  alarm?:Alarm,
  alarmId?:string
}
export interface ActivatedAlarmDTO{
  
  Timestamp:Date,
  Threshold:number,
  Message:string,
  Priority:number,
  Type:string,
  MeasureUnit:string
}
export interface AlarmPriorityDTO{
    type:string,
    timeOfActivation:Date,
    priority:number,
    tagName:string
}

export interface AlarmDTO{
  type:string,
  timeStamp:Date,
  priority:number,
  message:string,
  threshHold:number
}
export interface CreateAlarmDTO{
  analogId:number,
  threshold:number,
  message:string,
  priority:number,
  type:string
}

File: back/scada/scada/Models/Alarm.cs
﻿namespace scada.Models
{
    public class Alarm
    {
        public string Id { get; set; } 

        public int threshHold { get; set; }
        public string Message { get; set; }
        public AnalogInput analogInput { get; set; }

        public int? analogId { get; set; }

        public DateTime timeStamp { get; set; }
        public int priority { get; set; }
        public string Type { get; set; }

        public string? MeasureUnit { get; set; }
        public bool isDeleted { get; set; }
        public Alarm()
        {
            
        }

        public Alarm(int threshHold, string message, AnalogInput analogInput, int priority, string type)
        {
            this.threshHold = threshHold;
            this.Message = message;
            this.timeStamp = DateTime.Now;
            this.analogInput = analogInput;
            this.analogId = analogInput.id;
            this.priority = priority;
            Type = type;
            isDeleted = false;
        }
        public Alarm(int threshHold, string message, AnalogInput analogInput, int priority, string type, string measureUnit)
        {
            this.threshHold = threshHold;
            this.Message = message;
            this.timeStamp = DateTime.Now;
            this.analogInput = analogInput;
            this.analogId = analogInput.id;
            this.priority = priority;
            Type = type;
            MeasureUnit = measureUnit;

            isDeleted = false;
        }
        public Alarm(int threshHold, string message,  int priority, string type)
        {
            //this.analogId = analogInput.id;
            this.threshHold = threshHold;
            this.Message = message;
            this.timeStamp = DateTime.Now;
            this.priority = priority;
            Type = type;

            isDeleted = false;
        }
    }
}
